storm-core/src/jvm/org/apache/storm/scheduler/utils/ArtifactoryConfigLoader.java

```diff
@@ -0,0 +1,388 @@
+import org.apache.http.client.ResponseHandler;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.config.RequestConfig; 
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.impl.client.HttpClientBuilder; 
+import org.apache.http.client.HttpClient; 
+import org.apache.http.util.EntityUtils;
+import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.constructor.SafeConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ArtifactoryConfigLoader implements IConfigLoader {
+    /**
+     * Configuration items for this config loader are passed in via confg settings in 
+     * each scheduler that has a configurable loader.
+     *
+     * For example, the resource aware scheduler has configuration items defined in Config.java
+     * that allow a user to configure which implementation of IConfigLoader to use to load
+     * specific scheduler configs as well as any parameters to pass into the prepare method of
+     * tht configuration.
+     *
+     * resource.aware.scheduler.user.pools.loader can be set to org.apache.storm.scheduler.utils.ArtifactoryConfigLoader
+     *
+     * and then
+     *
+     * resource.aware.scheduler.user.pools.loader.params can be set to any of the following
+     *
+     *  {"artifactory.config.loader.uri": "http://artifactory.example.org:9989/artifactory/confs/my_cluster/mt_user_pool"}
+     *
+     *  {"artifactory.config.loader.uri": "file:///confs/my_cluster/mt_user_pool"}
+     *
+     *  {"artifactory.config.loader.uri": "file:///confs/my_cluster/mt_user_pool", "artifactory.config.loader.timeout.secs" : "60"}
+     **/
+    protected static final String ARTIFACTORY_URI = "artifactory.config.loader.uri";
+    protected static final String ARTIFACTORY_TIMEOUT_SECS="artifactory.config.loader.timeout.secs";
+    protected static final String ARTIFACTORY_POLL_TIME_SECS="artifactory.config.loader.polltime.secs";
+    protected static final String ARTIFACTORY_SCHEME="artifactory.config.loader.scheme";
+    protected static final String ARTIFACTORY_BASE_DIRECTORY="artifactory.config.loader.base.directory";
+    protected static final String LOCAL_ARTIFACT_DIR="scheduler_artifacts";
+    static final String cacheFilename = "latest.yaml";
+
+    private static final Logger LOG = LoggerFactory.getLogger(ArtifactoryConfigLoader.class);
+
+    @SuppressWarnings("rawtypes")
+    private Map _conf;
+    private int _artifactoryPollTimeSecs = 600;
+    private boolean _cacheInitialized = false;
+    // Location of the file in the artifactory archive.  Also used to name file in cache.
+    private String _localCacheDir;
+    private String _artifactoryScheme = "http";
+    private String _baseDirectory = "/artifactory";
+    private int _lastReturnedTime = 0;
+    private int _timeoutSeconds = 10;
+    private Map _lastReturnedValue;
+
+    private static class OurResponseHandler implements ResponseHandler<String> {
```

Comment:
Perhaps we can name this a bit better.  It appears to validate that the response code was in the 2xx range and then convert the body to a string.  Maybe `GETStringResponseHandler` would be better?